analytics.go

```diff
@@ -332,17 +330,25 @@ func (c *Client) loop() {
 				c.verbose("interval reached – nothing to send")
 			}
 		case <-c.quit:
-			tick.Stop()
 			c.verbose("exit requested – draining msgs")
-			// drain the msg channel.
+
```

Comment:
The flag would need to be atomic or protected by a lock, and there's always the risk of the state of that flag and the 'close' state of the channel to get out of sync if someone makes a change and didn't anticipate the side effects.
I feel like the channel here gives us all we need already, we would only need to recover from the panic in the enqueue method and return an error when it's already closed, plus it doesn't slow down the normal flow with extra synchronization work. 

On a side note we *have to* close the channel anyway to use a simple for loop to flush it.